hooks/_common.sh

```diff
@@ -219,10 +222,22 @@ function common::per_dir_hook {
 
   # Lookup hook-config for modifiers that impact common behavior
   local change_dir_in_unique_part=false
+  # Limit the number of parallel processes to the number of CPU cores -1
+  # `nproc` - linux, `sysctl -n hw.ncpu` - macOS, `echo 1` - fallback
+  local CPU
+  CPU=$(nproc || sysctl -n hw.ncpu || echo 1)
+  local parallelism_limit
+  local parallelism_disabled=false
+
   IFS=";" read -r -a configs <<< "${HOOK_CONFIG[*]}"
   for c in "${configs[@]}"; do
     IFS="=" read -r -a config <<< "$c"
-    key=${config[0]}
+
+    # $hook_config receives string like '--foo=bar; --baz=4;' etc.
+    # It gets split by `;` into array, which we're parsing here ('--foo=bar' ' --baz=4')
+    # Next line removes leading spaces, to support >1 `--hook-config` args
+    # shellcheck disable=SC2001 # Rule exception
+    key=$(sed 's/^[[:space:]]*//' <<< "${config[0]}")
```

Comment:
For Max's convenience:
```suggestion
    key="${config[0]## }"
```